src/main/java/com/nvs/store/models/cart/Cart.java

```diff
@@ -1,20 +1,61 @@
 package com.nvs.store.models.cart;
 
 import com.nvs.store.models.product.Product;
-import lombok.Data;
-import lombok.NonNull;
-import lombok.RequiredArgsConstructor;
+import jakarta.persistence.*;
 
-import java.math.BigDecimal;
-import java.util.List;
-
-@Data
-@RequiredArgsConstructor
+import java.util.Date;
 
+@Entity
+@Table(name = "carts")
 public class Cart {
-    @NonNull
-    private Long id;
-    private final List<Product> cartProducts;
-    @NonNull
-    private BigDecimal totalPrice;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    @Column(name = "created_date")
+    private Date createdDate;
+
+    @ManyToOne
+    @JoinColumn(name = "product_id")
+    private Product product;
```

Comment:
Defining relation as many to one between cart and product, is meaning that cart can include only single product. Please think how you can improve it to support add of many products.